core/network/fabric-interop-cc/libs/assetexchange/assetSwapContracts.go

```diff
@@ -1003,6 +1014,6 @@ func getHashPreImageHelper(ctx contractapi.TransactionContextInterface, key stri
 	if hashPreImageBase64Bytes == nil {
 		return "", logThenErrorf("key %s is not associated with any claimed asset", key)
 	}
-	
+
 	return string(hashPreImageBase64Bytes), nil
```

Comment:
Shouldn't we return a structure with the hash mechanism here, just like we do in `getHTLCHashHelper`?

Reply:
As discussed, since this is the preimage pertaining to just the hash mechanism ( and not generic to all lock mechanisms), we will not include the hash mechanism in the return value (and assume that the caller already knows this).